src/jobsautoreport/slack.py

```diff
@@ -14,87 +14,126 @@ def __init__(self, web_client: WebClient, channel_id: str) -> None:
         self._client = web_client
         self._channel_id = channel_id
 
-    def send_report(self, report: Report) -> None:
-        blocks = self._format_message(report)
-        logger.debug("Message format finished successfully")
+    def _post_message(
+        self,
+        report: Report,
+        format_function: Callable[[Report], list[dict[str, Any]]],
+        thread_time_stamp: Optional[str],
+    ) -> str:
+        blocks = format_function(report)
+        logger.debug("Message formatted successfully")
         response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks
+            channel=self._channel_id, blocks=blocks, thread_ts=thread_time_stamp
         )
         response.validate()
         logger.info("Message sent successfully")
+        return response["ts"]
 
-        time_stamp = response["ts"]
-        blocks = self._format_presubmit_comment(report)
-        logger.debug("comment format finished successfully")
-        response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks, thread_ts=time_stamp
+    def _create_and_upload_file(
+        self,
+        jobs: list[tuple[str, Any]],
+        file_title: str,
+        thread_time_stamp: Optional[str],
+        top_triggered: bool,
+    ) -> None:
+        filename = file_title.replace(" ", "_").lower()
+        file_path = f"/tmp/{filename}.png"
+        initial_comment = file_title
+        if top_triggered:
+            self._create_quantity_image(file_path, jobs)
+        else:
+            self._create_success_image(file_path, jobs)
```

Comment:
A generic function like ``_create_and_upload_file`` shouldn't be customized by arguments (``top_triggered``) to do specific actions (``create_quantity_image`` and ``create_success_image``) but rather be used by them